tests/constraints/test_cardinality_constraint_discrete.py

```diff
@@ -0,0 +1,51 @@
+"""Test for the discrete cardinality constraint."""
+
+from itertools import combinations
+
+import numpy as np
+import pytest
+
+from baybe.constraints.discrete import DiscreteCardinalityConstraint
+from baybe.parameters.numerical import NumericalDiscreteParameter
+from baybe.searchspace import SearchSpace
+
+# Combinations of cardinalities to be tested
+_cardinalities = sorted(combinations(range(0, 5), 2))
+
+
+@pytest.mark.parametrize(
+    "cardinalities", _cardinalities, ids=[str(x) for x in _cardinalities]
+)
+def test_cardinality_constraint_discrete(cardinalities, n_grid_points):
```

Comment:
Would appreciate type hints.

Reply:
Also, the naming is a bit confusing (also in the other test): `_cardinalities` is a collection of cardinalities, but here, `cardinalities` is a pair (`min_cardinality`, `max_cardinality`).

Reply:
I specify the names now to 
- `_cardinalities` -> `cardinality_bounds_combinations`
- `cardinalities` -> `cardinality_bounds`